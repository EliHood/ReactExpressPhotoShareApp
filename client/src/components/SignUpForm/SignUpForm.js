import React  from "react";
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
const SignUpForm = (props) => {

    return(
        <form onSubmit={props.signSubmit}>
            <TextField
                label="Username"
                style={{width: '100%' }}
                name="username"
                value={props.username}
                onChange={props.usernameChange}
                margin="normal"
                />
            <br></br>
            <TextField
                label="Email"
                className=""
                style={{width: '100%' }}
                name="email"
                value={props.email}
                onChange={props.handleEmail}
                margin="normal"
            />
            <br></br>
            <TextField
                label="Password"
                name="password"
                type="password"
                helperText={props.password_error_text }
                style={{width: '100%' }}
                error={props.passErr}
                className=""
                value={props.password}
                onChange={props.handlePassword}
                margin="normal"
                />
                {/*  */}
            <br></br>
            <TextField
                label="Confirm Password"
                name="passwordConf"
                type="password"
                helperText={props.passwordConf_error_text }
                error={props.passwordConfpassErr}
                style={{width: '100%' }}
                className=""
                value={props.passwordConf}
                onChange={props.handlePasswordConf}
                margin="normal"
            />
            <br></br>
            <br></br>
        
            <Button 
                disabled={props.validation}  
                variant="outlined" color="primary" type="submit">
                Sign Up
            </Button>
        </form>  
    )
  
}

export default SignUpForm; 