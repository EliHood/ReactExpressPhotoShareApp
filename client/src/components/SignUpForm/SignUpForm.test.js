import React from 'react';
import { shallow } from 'enzyme';
import Button from '@material-ui/core/Button';
import toJson from 'enzyme-to-json';
import SignUpForm from './SignUpForm';

describe('should render <SignUpForm/>', () => {
  it('should render SignUpForm', () => {
    const wrapper = shallow(<SignUpForm />);
    expect(wrapper.find('form')).toHaveLength(1);
  });

  it('should match SignUpForm snap', () => {
    const wrapper = shallow(<SignUpForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Button', () => {
    const wrapper = shallow(<SignUpForm />);
    expect(wrapper.find(Button)).toHaveLength(1);
  });
});
