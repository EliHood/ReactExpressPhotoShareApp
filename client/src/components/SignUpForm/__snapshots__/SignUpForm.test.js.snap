// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <SignUpForm/> should match SignUpForm snap 1`] = `
<form>
  <WithStyles(ForwardRef(TextField))
    label="Username"
    margin="normal"
    name="username"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <br />
  <WithStyles(ForwardRef(TextField))
    className=""
    label="Email"
    margin="normal"
    name="email"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <br />
  <WithStyles(ForwardRef(TextField))
    className=""
    label="Password"
    margin="normal"
    name="password"
    style={
      Object {
        "width": "100%",
      }
    }
    type="password"
  />
  <br />
  <WithStyles(ForwardRef(TextField))
    className=""
    label="Confirm Password"
    margin="normal"
    name="passwordConf"
    style={
      Object {
        "width": "100%",
      }
    }
    type="password"
  />
  <br />
  <br />
  <WithStyles(ForwardRef(Button))
    color="primary"
    type="submit"
    variant="outlined"
  >
    Sign Up
  </WithStyles(ForwardRef(Button))>
</form>
`;
